### PR TITLE
feat: Add SENTRY_SPOTLIGHT env variable support

### PR DESCRIPTION
**Pull request by BYK**
Allows setting Spotlight through `$SENTRY_SPOTLIGHT` env variable.

Part of getsentry/spotlight#482

### CODE REVIEW

File: sentry_sdk/client.py [hunk 1]
```diff
@@ -375,6 +375,16 @@ def _capture_envelope(envelope):
             )
 
             self.spotlight = None
+            spotlight_config = self.options.get("spotlight")
+            if spotlight_config == None:
```
Comment (sentrivana) [inline sentry_sdk/client.py:379]:
```suggestion
            if spotlight_config is None:
```

**Review by sentrivana — COMMENTED** [review]
@BYK Thanks for this! Can you please also add a test to https://github.com/getsentry/sentry-python/blob/48589966945785787a2855533386a2648e9df784/tests/test_spotlight.py?

File: sentry_sdk/client.py (hunk 1 above)
Comment (sentrivana) [reply sentry_sdk/client.py:379]: This should also fix the CI / Lint Sources workflow

**Comment by BYK** [comment]
Will add tests and then will request a re-review.

**Comment by BYK** [comment]
Finally, ready for the final round

**Review by sentrivana — APPROVED** [review]
Thank you @BYK!

**Pull request merged**